src/Marten/Schema/FullTextIndex.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Baseline;
+using Marten.Storage;
+
+namespace Marten.Schema
+{
+    public class FullTextIndex : IIndexDefinition
+    {
+        private readonly string _config;
+        private readonly DbObjectName _table;
+        private string _indexName;
+
+        public FullTextIndex(DocumentMapping parent, string config)
+        {
+            _table = parent.Table;
+            _config = config;
+        }
+
+        public string IndexName
+        {
+            get
+            {
+                if(_indexName.IsNotEmpty())
```

Comment:
What I'd suggest is to move this logic of checking and parsing to setter method. Setter method will be called less, so that will reduce the number of potential allocations